app/src/main/java/co/icanteach/projectx/data/NetworkModule.kt

```diff
@@ -11,6 +13,7 @@ import retrofit2.create
 import javax.inject.Singleton
 
 @Module
+@InstallIn(ApplicationComponent::class)
 class NetworkModule {
```

Comment:
instead of class, using object is better for here from the performance point of view.